replica/compaction_group.hh

```diff
@@ -248,6 +259,9 @@ public:
         return _compaction_groups.size() == 1 ? &_compaction_groups.front() : nullptr;
     }
 
+    // Deep copy of the instance.
+    // Used for taking a snapshot of the table's storage_groups
+    virtual std::unique_ptr<storage_group_manager> clone() const = 0;
```

Comment:
Why do we clone the entire manager stuff? Seems overkill. We just need a vector of sstable_sets + range definitions.